Iniciante/1957/README.md

```diff
@@ -0,0 +1,86 @@
+# Problema:
+Os dados armazenados no computador estão em binário. Uma forma econômica de ver estes números é usar a base 16 (hexadecimal).
+
+Sua tarefa consiste em escrever um programa que, dado um número natural na base 10, mostre sua representação em hexadecimal.
+
+**Problema Completo**: https://www.urionlinejudge.com.br/judge/pt/problems/view/1957
+
+
+# Resolução:
+O objetivo deste exercício é bem simples: transformar um número que está 10 para a base 16. Portanto, é importante relembrarmos a correspondência destas:
+Base 10 | Base 16
+--------|---------
+   0    |    0
+   1    |    1
+   2    |    2
+   3    |    3
+   4    |    4
+   5    |    5
+   6    |    6
+   7    |    7
+   8    |    8
+   9    |    9
+   10   |    A
+   11   |    B
+   12   |    C
+   13   |    D
+   14   |    E
+   15   |    F
+   16   |    10
+   17   |    11
+  ...   |    ...
+
+Visto que a base 16 inclui caracteres (strings), é essencial a importação da biblioteca `string.h` para possibilitar seu uso. Além disso, com o intuito de instanciar uma variável sem determinar seu tamanho no momento da declaração (ou seja, trabalhar com alocação dinâmica), utilizaremos uma função inclusa na biblioteca `stdlib.h`.  
+
+```c
+#include <stdlib.h>
+#include <string.h>
+```  
+
+Como primeiro passo, declaramos a variável inteira `V`, na qual será guardado, através da função `scanf()`, o número fornecido (pertencente à base 10).  
+
+```c
+int V;
+scanf("%d",&V);
+```  
+
+Observando a tabela anterior, podemos notar que os valores em hexadecimal possuem apenas 16 tipos diferentes de caracteres: 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, A, B, C, D, E e F. Tendo esta informação em mãos, iremos declarar `digitosHexa` para armazená-los e, sabendo que todo vetor de `char` possui como último elemento o caractere especial `\0`, este possuirá tamanho 17.
+Também é importante declararmos o inteiro `i`, para auxiliar nos cálculos; e `Vhexa`, que conterá o resultado obtido. Porém, teremos conhecimento do tamanho da string gerada em hexadecimal apenas durante a execução do programa e, consequentemente, não há como alocar, com antecedência, a quantidade de memória necessária. Em cenários como este, podemos desfrutar da função `malloc` para realizar uma alocação dinâmica, sendo necessário fornecer-lhe somente o tamanho do tipo a ser utilizado (neste caso, `char`).  
```

Comment:
```suggestion
Também é importante declararmos o inteiro `i`, para auxiliar na iteração do `for`; e `Vhexa`, que conterá o resultado obtido. Porém, teremos conhecimento do tamanho da string gerada em hexadecimal apenas durante a execução do programa e, consequentemente, não há como alocar, com antecedência, a quantidade de memória necessária. Em cenários como este, podemos desfrutar da função `malloc` para realizar uma alocação dinâmica, sendo necessário fornecer-lhe somente o tamanho do tipo a ser utilizado (neste caso, `char`).  
```

Reply:
O `i`não é usado só na iteração do `for`

Reply:
É que você tá usando ele como indice de `Vhexa` e para poder imprimir `Vhexa` ao contrário, isso já dentro do `for. Acho que poderia ter algo assim, por que auxiar nos cálculos é algo muito abrangente, pra mim pelo menos, e também não tem operação do `i`com outras variáveis, então achei um pouco estranho.
Mudei a minha sugestão em cima